src/vm/methodtable.cpp

```diff
@@ -3226,7 +3226,7 @@ void MethodTable::AllocateRegularStaticBoxes()
     if (pClassCtorInfoEntry != NULL)
     {
         OBJECTREF* pStaticSlots = (OBJECTREF*)(pStaticBase + pClassCtorInfoEntry->firstBoxedStaticOffset);
-        GCPROTECT_BEGININTERIOR(pStaticSlots);
+        GCPROTECT_BEGININTERIOR(*pStaticSlots);
```

Comment:
This is bad change. Discussed in https://github.com/dotnet/coreclr/issues/23842

Reply:
Is there a way to tweak the code to make the analysis stop complaining about it?

Reply:
Ah, didn't noticed the discussion. Should we suppress `-Wsizeof-pointer-div` in CMakeLists?

Reply:
The other way is to inline the suppression: `#pragma clang diagnostic ignored "-Wsizeof-pointer-div"` (and the other one for gcc).

Reply:
It would be best to tweak the code to make the warning go away, e.g. add extra cast or something similar. What is the exact pattern that the analysis looks for?

Reply:
It was looking for the exact `sizeof(x)/sizeof(x[0])` vulgar fraction pattern in both clang v8 and gcc v8. I have extracted the numerator variable in a separate variable and confirmed that it recompiles with both. Thanks!